rustler/src/types/primitive.rs

```diff
@@ -60,3 +59,34 @@ impl<'a> Decoder<'a> for bool {
         atom::decode_bool(term)
     }
 }
+
+impl Encoder for f32 {
+    fn encode<'a>(&self, env: Env<'a>) -> Term<'a> {
+        #[allow(clippy::cast_lossless)]
```

Comment:
Is this needed because of `*self as f64`? If so, `f64::from(*self)` would probably be preferable.